### PR TITLE
Desugar interop method gen

### PR DESCRIPTION
**Pull request by lankavitharana**
Fixes #<Issue Number>

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by Kishanthan** [comment]
@lankavitharana , there are so manu json files changes in this PR which I believe coming from langserver. What was the reason?

**Comment by lankavitharana** [comment]
@Kishanthan These are language server test related jsons, this happened because I have added a dependency to bir module from jvm module, @nadeeshaan can explain the need of json better

File: compiler/ballerina-backend-jvm/src/main/ballerina/src/compiler_backend_jvm/interop/external_method_gen.bal [hunk 1]
```diff
@@ -67,6 +67,9 @@ function injectDefaultParamInits(bir:Package module) {
             if extFuncWrapper is OldStyleExternalFunctionWrapper {
                 desugarOldExternFuncs(module, extFuncWrapper, birFunc);
                 enrichWithDefaultableParamInits(birFunc);
+	        } else if (extFuncWrapper is JMethodFunctionWrapper) {
```
Comment (Kishanthan) [inline compiler/ballerina-backend-jvm/src/main/ballerina/src/compiler_backend_jvm/interop/external_method_gen.bal]: formatting issue

File: compiler/ballerina-backend-jvm/src/main/ballerina/src/compiler_backend_jvm/jvm_error_gen.bal [hunk 2]
```diff
@@ -37,13 +37,13 @@ type ErrorHandlerGenerator object {
         self.mv.visitInsn(ATHROW);
     }
 
-    function generateTryInsForTrap(bir:ErrorEntry currentEE, string previousTargetBB, jvm:Label endLabel,
-                                   jvm:Label handlerLabel, jvm:Label otherErrorLabel, jvm:Label jumpLabel) {
-        jvm:Label startLabel = new;
+    function generateTryInsForTrap(bir:ErrorEntry currentEE, string previousTargetBB, jvm:Label startLabel, jvm:Label jumpLabel) {
+	    if (currentEE is JErrorEntry) {
```
Comment (Kishanthan) [inline compiler/ballerina-backend-jvm/src/main/ballerina/src/compiler_backend_jvm/jvm_error_gen.bal]: formatting issue

File: compiler/ballerina-backend-jvm/src/main/ballerina/src/compiler_backend_jvm/jvm_error_gen.bal [hunk 3]
```diff
@@ -59,10 +59,38 @@ type ErrorHandlerGenerator object {
         }
     }
 
-    function generateCatchInsForTrap(bir:ErrorEntry currentEE, jvm:Label endLabel,
-                                    jvm:Label errorValueLabel, jvm:Label otherErrorLabel, jvm:Label jumpLabel) {
+    function generateCatchInsForTrap(bir:Function func, bir:ErrorEntry currentEE, jvm:Label startLabel, jvm:Label endLabel, jvm:Label jumpLabel, 
+    						InstructionGenerator instGen, TerminatorGenerator termGen) {
         self.mv.visitLabel(endLabel);
         self.mv.visitJumpInsn(GOTO, jumpLabel);
+	    if (currentEE is JErrorEntry) {
```
Comment (Kishanthan) [inline compiler/ballerina-backend-jvm/src/main/ballerina/src/compiler_backend_jvm/jvm_error_gen.bal]: formatting issue in this whole section. looks like it does not use 4 columns space?

**Comment by Kishanthan** [comment]
I will go ahead and approve this. @lankavitharana can you send another PR with all the formatting issues fixed?

**Comment by lankavitharana** [comment]
@Kishanthan sure, will send a separate PR with formatting fixes

**Pull request merged**